DEVELOPING.md

```diff
@@ -0,0 +1,45 @@
+# Developing ros-cross-compile
+
+This document is intended to contain instructions, guidelines, and requirements for developing this tool.
+
+## Writing Code
+
+It is recommended to use `virtualenv` to manage the dependency environment.
+
+```
+# create a virtualenv
+virtualenv venv
+# use it
+source venv/bin/activate
+# install this package in the virtualenv to get dependencies
+pip install -e .
+# now you can run your work
+ros_cross_compile
+```
+
+## Tests
+
+There are currently two test entrypoints
+
+* Unit tests via `tox`
+  * run `tox -e py`
+* End-to-end shell script
+  * run `./test/run_e2e_test.sh`
+
+## Host Platforms
+
+This tool aims to support as host platforms
+* all ROS 2 Tier 1 platforms
+* on x86_64 (cross-compiling _from_ ARM platforms has not been deemed necessary)
+* for LTS distributions that have not hit EOL
+
+See [REP 2000](https://www.ros.org/reps/rep-2000.html) to determine this target, which as of this writing is:
+* Dashing Diademata
+  * Ubuntu Bionic 18.04
+    * support installation via `pip` and `apt`
```

Comment:
Is apt still recommended?

Reply:
Yes, Tully recommended that we release via apt so that this tool has a similar surface area to e.g. `colcon` and `rosdep`. I am convinced by that